### PR TITLE
enhance n_clusters

### PR DESCRIPTION
**Pull request by strengejacke**

### CODE REVIEW

**Comment by strengejacke** [comment]
Can you add the pkg to the DESCRIPTION as well?

**Comment by DominiqueMakowski** [comment]
Not really, it's not on CRAN 😱

**Comment by strengejacke** [comment]
Will that ever pass any check?

**Comment by DominiqueMakowski** [comment]
Well yes because we don't use it in the EXAMPLE 😏

**Comment by DominiqueMakowski** [comment]
we'll see, worst case scenario we remove it

**Comment by strengejacke** [comment]
> The ‘Additional_repositories’ field is a comma-separated list of repository URLs where the packages named in the other fields may be found. It is currently used by R CMD check to check that the packages can be found, at least as source packages (which can be installed on any platform).

**Comment by strengejacke** [comment]
We may add 

`Additional_repositories: https://bioconductor.org/packages/release/bioc/html/M3C.html`

to the DESCRIPTION file?

**Comment by DominiqueMakowski** [comment]
I didn't know this field existed, and wonder how is it different from "Remotes". But it sounds appropriate indeed

**Comment by strengejacke** [comment]
I'LL ask at r-package-devel (https://stat.ethz.ch/mailman/listinfo/r-package-devel), but currently, I always get a timeout when I try to subscribe...

**Comment by strengejacke** [comment]
haha, that happens when R relies on Python... The list is

image

so no wonder it doesn't work properly ;-)

**Comment by strengejacke** [comment]
Ok, I got following answer:

> BioC is a standard repository that you do not need to delcare.

So, we can/should be safe to add M3C to "Suggest".

**Comment by DominiqueMakowski** [comment]
does it mean that it knows automatically that if it's not from CRAN it needs to be downloaded from bioconductor?

**Comment by strengejacke** [comment]
> does it mean that it knows automatically that if it's not from CRAN it needs to be downloaded from bioconductor?

Yes, this is how I understood Uwe.

**Comment by DominiqueMakowski** [comment]
Interesting

File: R/format_p.R [hunk 1]
```diff
@@ -26,7 +26,7 @@ format_p <- function(p, stars = FALSE, stars_only = FALSE, name = "p", missing =
       ifelse(p < 0.001, "< .001***",
         ifelse(p < 0.01, "< .01**",
           ifelse(p < 0.05, "< .05*",
-            paste0("= ", insight::format_value(p, digits))
+            paste0("= ", insight::format_value(p, 3))
```
Comment (strengejacke) [inline R/format_p.R]: Why this change?

Comment (strengejacke) [reply R/format_p.R]: ok, I assume this is an APA requirement?

Comment (DominiqueMakowski) [reply R/format_p.R]: Otherwise it fails because digits is a character ("apa") and format_value expects an int

**Comment by strengejacke** [comment]
Dear pull-requester,
see the errors related to your current PR. Please fix asap, latest before 8. Feb. 2020 to safely retain your PR on GitHub.
Sincerly,
easystats maintainance

**Comment by DominiqueMakowski** [comment]
I knew it! @strengejacke was a bot this whole time 🤖

**Comment by DominiqueMakowski** [comment]
I've been so used to seeing parameters, performance and insight fail travis that I didn't even think it could come from my commit 😬

**Comment by strengejacke** [comment]
> I've been so used to seeing parameters, performance and insight fail travis

Wait, that must have been **years** ago. insight is fine, parameters as well, and performance usually, too. It currently fails because it needs the latest _parameters_ on CRAN (which I plan to submit next week, because I need to submit performance to CRAN).

**Comment by strengejacke** [comment]
I think you need to merge master into dev, and then add M3C as suggested, and that's it.

**Comment by DominiqueMakowski** [comment]
There was a missing closing quote in my commit as well 🙄  fixing now

**Comment by strengejacke** [comment]
when this PR passes, I'll merge and include it in the next update, or not? can you also update the news, please? 😬

**Comment by DominiqueMakowski** [comment]
>  I'll merge and include it in the next update

Yes, yes, necessary for report!
